src/main/java/org/folio/services/item/ItemService.java

```diff
@@ -109,6 +109,16 @@ public ItemService(Context vertxContext, Map<String, String> okapiHeaders) {
     return itemRepository.getById(itemId)
       .compose(CommonValidators::refuseIfNotFound)
       .compose(oldItem -> refuseWhenHridChanged(oldItem, newItem))
+      .compose(oldItem -> {
+        var oldDestServicePointId = oldItem.getInTransitDestinationServicePointId();
+        if (oldDestServicePointId != null &&
+          newItem.getInTransitDestinationServicePointId() == null) {
+
+          newItem.setInTransitDestinationServicePointId(oldDestServicePointId);
+        }
+
+        return succeededFuture(oldItem);
+      })
```

Comment:
This suggests that the in transit destination service point should be retained even if it is removed by the client making a change.

Why is this needed?

Does this make it impossible for a client to remove the destination service point?

Reply:
Thanks @marcjohnson-kint, you are right.
When an item is becoming "Available" mod-circulation on check-in sends InTransitDestinationServicePointId == null.
I've updated the condition and now the fix should work correctly for such case.

Another solution can be implemented on UI side: check InTransitDestinationServicePointId of old item and if nothing changes, send this field in new item.

Reply:
> When an item is becoming "Available" mod-circulation on check-in sends InTransitDestinationServicePointId == null.
> I've updated the condition and now the fix should work correctly for such case.

Has it been decided by a PO that if an item is In-Transit it should *always* retain the destination service point irrespective of whether a client wants to remove it?

> Another solution can be implemented on UI side: check InTransitDestinationServicePointId of old item and if nothing changes, send this field in new item.

Personally, unless this is a deliberate rule, I think this would be better resolved in the client, which is the UI.

I think this kind of logic can easily lead to folks being surprised when they think they've removed a property from a record representation and instead, as if by magic, it comes back.

Reply:
Thanks @marcjohnson-kint, we discussed it with FE team, it will be fixed on UI side. I'll close this PR